numpy/ma/extras.py

```diff
@@ -592,7 +592,7 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.count(axis))
     else:
-        wgt = np.asanyarray(weights)
+        wgt = asarray(weights)
```

Comment:
I think this could have side effects for things like cupy arrays. Is it required?

Reply:
Yes because later I add a mask to the weights array.

Reply:
I see there are many more instances of `asarray` than of `asanyarray` across the masked array implementation. So this problem is not unique to this PR.